modules/sdk/src/main/scala/org/tessellation/sdk/http/routes/SnapshotRoutes.scala

```diff
@@ -91,6 +91,21 @@ final case class SnapshotRoutes[F[_]: Async: KryoSerializer, S <: Snapshot[_, _]
           }
         }.getOrElse(NotFound())
 
+    case GET -> Root / "hash" / SnapshotOrdinalVar(ordinal) :? FullSnapshotQueryParam(fullSnapshot) =>
```

Comment:
What about the NotFound http status that was there before?

Reply:
The route is returning an `Option[Hash]` now. So, if it doesn't find the hash for the snapshot at the specified `SnapshotOrdinal`, it returns an `Ok(None)`. There is no scenario where it will return a `NotFound`.

Reply:
Why does it return Some/None instead of Ok and NotFound status codes? IMO if there is no resource, the endpoint should return NotFound
@AlexBrandes what do you think?

Reply:
The route is now returning `Ok(Hash)` or `NotFound`. `L0GlobalSnapshotClient` converts successful calls to `Some(Hash)` and failures to `none[Hash]`.

Reply:
Great. All failures or just the NotFound status code?

Reply:
All failures.

Reply:
Is it correct behavior? Shouldn't anything else throw to stop the algorithm or something? If no,and it's correct then we're good.

Reply:
This behavior was implemented because as @sdavidp and I were tracing `PeerResponse`, we were unsure of how it handled errors. We thought that it might be converting all errors into `unauthenticated` throwables. Seeing as we didn't want to modify `PeerResponse` or the middleware it uses for the actual call, we simply converted all errors to `None`.
Part of the rationale for doing that was also the thought that if any error was encountered, a `none[Hash]` makes sense as a return value.

Reply:
I think the easiest would be to talk with protocol_eu and make quick pair programming. Anyway, for me personally, "none"/"not found" sounds like a domain error (cannot find a snapshot) but I know the request was correctly processed. If I have an error, I can retry the request if needed or do some other logic. Right now we're not able to distinguish whether the request was incorrect or the node has no snapshot. If you're good with that then fine but in my opinion it should reflect the real state (found, not found, issue with the request/response).

Reply:
> Right now we're not able to distinguish whether the request was incorrect or the node has no snapshot. If you're good with that then fine but in my opinion it should reflect the real state (found, not found, issue with the request/response).

Yes, this is why we were asking about where the 404 was mapped to an error in `F` so we could distinguish between that and real errors. I still haven't found it in _http4s_ or our code.

Reply:
Only `Status.NotFound` and `Status.None` return `none[Hash]` now.

Reply:
You mean Status.Gone? :) 
Cool, I'll take a look tomorrow morning and approve/merge